services/klatchat_observer/controller.py

```diff
@@ -728,6 +742,37 @@ def on_subminds_state(self, body: dict):
         body["msg_type"] = "subminds_state"
         self.sio.emit("broadcast", data=body)
 
+    @create_mq_callback()
+    def on_get_configured_personas(self, body: dict):
+        response_data = self._fetch_persona_api(body=body)
+        self.send_message(
+            request_data=response_data,
+            vhost=self.get_vhost("llm"),
+            queue="get_configured_personas.response",
+            expiration=3000,
+        )
+
+    def _fetch_persona_api(self, body: dict) -> dict:
+        query_string = self._build_persona_api_query(body=body)
+        url = f"{self.server_url}/personas/list?{query_string}"
+        response = self._fetch_klat_server(url=url)
+        if response.ok:
+            data = response.json()
+        else:
+            LOG.error(
+                f"Failed to fetch personas from url {url}, server responded with status {response.status_code}"
+            )
+            data = {"items": []}
+        return data
+
+    def _build_persona_api_query(self, body: dict) -> str:
+        url_query_params = []
+        if service_name := body.get("service_name"):
+            url_query_params.append(f"llm={service_name}")
```

Comment:
> can/should `service_name` be used in the URL instead of `llm`? If those parameters are always equivalent, I think it makes sense for them to have the same name.

naming it "service_name" could be a bit misleading, as intention is to gather supported personas according to requesting llm & requested user id

Reply:
> can/should `service_name` be used in the URL instead of `llm`? If those parameters are always equivalent, I think it makes sense for them to have the same name.

I think its better to use the work "llm" as we are here querying personas of particular llm

Reply:
and from perspective of LLM that emits that request this parameter is associated with its name, thus "service_name"